### PR TITLE
Task70

### PR DESCRIPTION
**Pull request by olgamorozova1**

### CODE REVIEW

File: src/main/java/by/issoft/training/utils/Converter.java [hunk 1]
```diff
@@ -54,16 +55,24 @@ private static String convertObjectToStringFormat(Object object, ObjectMapper ma
...
+            mapper.writeValue(byteArrayOutputStream, list);
             byte[] data = byteArrayOutputStream.toByteArray();
             json = new String(data, StandardCharsets.UTF_8);
         } catch (IOException ioException) {
             ioException.printStackTrace();
         }
         return json;
     }
+
+    public static <T> void convertListToJsonFile(List<T> list, String pathname) {
```
Comment (EugenBorisik) [inline src/main/java/by/issoft/training/utils/Converter.java:71]: Change to wildcard:
`public static void convertListToJsonFile(List<?> list, String pathname)`

File: src/main/java/by/issoft/training/utils/Converter.java [hunk 2]
```diff
@@ -54,16 +55,24 @@ private static String convertObjectToStringFormat(Object object, ObjectMapper ma
         return listOfObjects;
     }
 
-    public static String convertListOfStringsToJson(List<String> listOfStrings) {
+    public static <T> String convertListToJson(List<T> list) {
```
Comment (EugenBorisik) [inline src/main/java/by/issoft/training/utils/Converter.java:58]: Remove this method, use convertObjectToJson where needed

File: bug_reports/BugReportsTask70 [hunk 3]
```diff
@@ -0,0 +1,46 @@
+1. 500 error is received when post upload request with header Content-type: "multipart/form-data" is sent.
```
Comment (EugenBorisik) [inline bug_reports/BugReportsTask70:1]: This bug doesn't exist

File: src/main/java/by/issoft/training/clients/UserClient.java [hunk 4]
```diff
@@ -71,4 +74,20 @@ public CloseableHttpResponse deleteUser(UserDto user, String contentType) {
         }
         return delete.executeRequest("/users");
     }
+
+    public <T> HttpResponse uploadUsers(List<T> listOfObjects) {
```
Comment (EugenBorisik) [inline src/main/java/by/issoft/training/clients/UserClient.java:78]: Change to wildcard:
`public HttpResponse uploadUsers(List<?> listOfObjects)`

File: src/main/java/by/issoft/training/requests/Post.java [hunk 5]
```diff
@@ -37,4 +45,24 @@ public CloseableHttpResponse executeRequest(String path) {
...
+        HttpResponse uploadResponse = null;
+        try {
+            HttpClient httpclient = HttpClients.createDefault();
+            HttpPost postRequest = new HttpPost(readInfoFromProperties("url") + path);
+            postRequest.setHeader(HttpHeaders.AUTHORIZATION, authHeader);
+            MultipartEntityBuilder builder = MultipartEntityBuilder.create();
+            builder.setMode(HttpMultipartMode.BROWSER_COMPATIBLE);
+            String pathname = "./target/objects.json";
+            convertListToJsonFile(listOfObjects, pathname);
+            builder.addBinaryBody("file", new File(pathname));
```
Comment (EugenBorisik) [inline src/main/java/by/issoft/training/requests/Post.java:59]: multipart/form-data works:
`File file = new File(pathname);
 builder.addBinaryBody("file", file, ContentType.MULTIPART_FORM_DATA, file.getName());`

File: src/main/java/by/issoft/training/requests/Post.java [hunk 6]
```diff
@@ -37,4 +45,24 @@ public CloseableHttpResponse executeRequest(String path) {
         }
         return response;
     }
+
+    public <T> HttpResponse executeUploadRequest(String path, List<T> listOfObjects) {
```
Comment (EugenBorisik) [inline src/main/java/by/issoft/training/requests/Post.java:49]: Use wildcard:
`public HttpResponse executeUploadRequest(String path, List<?> listOfObjects)`

File: src/main/java/by/issoft/training/test/Task70UploadUsersTest.java [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static by.issoft.training.utils.StringGenerator.generateUserName;
+import static by.issoft.training.utils.StringGenerator.generateZipCode;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class Task70UploadUsersTest extends BaseTest {
```
Comment (EugenBorisik) [inline src/main/java/by/issoft/training/test/Task70UploadUsersTest.java:16]: Add one more test for 400 response code

**Review by EugenBorisik — CHANGES_REQUESTED** [review]

File: src/main/java/by/issoft/training/utils/Converter.java (hunk 1 above)
Comment (olgamorozova1) [reply src/main/java/by/issoft/training/utils/Converter.java]: Fixed

File: src/main/java/by/issoft/training/utils/Converter.java (hunk 2 above)
Comment (olgamorozova1) [reply src/main/java/by/issoft/training/utils/Converter.java]: Removed

File: bug_reports/BugReportsTask70 (hunk 3 above)
Comment (olgamorozova1) [reply bug_reports/BugReportsTask70]: Not a bug, deleted

Comment (olgamorozova1) [reply bug_reports/BugReportsTask70]: Not a bug, deleted

File: src/main/java/by/issoft/training/clients/UserClient.java (hunk 4 above)
Comment (olgamorozova1) [reply src/main/java/by/issoft/training/clients/UserClient.java]: Fixed

File: src/main/java/by/issoft/training/requests/Post.java (hunk 5 above)
Comment (olgamorozova1) [reply src/main/java/by/issoft/training/requests/Post.java:59]: Discussed, no changes are needed.

File: src/main/java/by/issoft/training/requests/Post.java (hunk 6 above)
Comment (olgamorozova1) [reply src/main/java/by/issoft/training/requests/Post.java]: Fixed

File: src/main/java/by/issoft/training/test/Task70UploadUsersTest.java (hunk 7 above)
Comment (olgamorozova1) [reply src/main/java/by/issoft/training/test/Task70UploadUsersTest.java:16]: Added new test.

**Review by EugenBorisik — APPROVED** [review]

**Pull request merged**